privacyidea/lib/crypto.py

```diff
@@ -863,7 +865,7 @@ def generate_password(size=6, characters=string.ascii_lowercase +
     passwd = [urandom.choice(str) for str in requirements]
     # fill the password until size with allowed characters
     passwd.extend(urandom.choice(characters) for _x in range(size - len(requirements)))
-    # return shuffled password
+    # return shuffled password - as the password was already randomly generated, this might actually be not necessary
```

Comment:
I think in this case the `shuffle` is required since we first add the characters from the required classes and then fill the list up with characters from the default classes.

Reply:
You are right. Will shuffle again.

Reply:
You are right. This differs from the first occurance.

Reply:
Done d4903c38